content/calloop-v0.7.markdown

```diff
@@ -0,0 +1,121 @@
+not shareable between threads. Calloop is meant to run the event loop on a single thread.
+
+The `EventSource` trait is a generic interface for "something that can generate events". Anything
+implementing that trait can be associated to a callback in the `EventLoop`. It will be monitored
+and the associated callback will be invoked every time the event source generates an event. This
+trait is meant to be composable: one can implement a high-level event source by wrapping a low-level
+one and further processing its event before passing them along.
+
+To give you a feel of what it looks like, take a look at this example extracted from the documentation:
+
+```rust
+extern crate calloop;
+
+use calloop::{generic::Generic, EventLoop, Interest, Mode};
+
+use std::time::Duration;
+
+fn main() {
+    // Create the event loop
+    let mut event_loop = EventLoop::try_new()
+                .expect("Failed to initialize the event loop!");
+    // Retrieve an handle. It is used to insert new sources into the event loop
+    // It can be cloned, allowing you to insert sources from within source callbacks
+    let handle = event_loop.handle();
+
+    // Inserting an event source takes this general form
+    // it can also be done from within the callback of an other event source
+    handle.insert_source(
+        // a type implementing the EventSource trait
+        source,
+        // a callback that is invoked whenever this source generates an event
+        |event, metadata, shared_data| {
+            // This callback is given 3 values:
+            // - the event generated by the source
+            // - &mut access to some metadata, specific to the event source
+            // - &mut access to the global shared data that was passed to EventLoop::dispatch
+        }
+    );
+
+    // Actual run of your loop
+    //
+    // Dispatch received events to their callbacks, waiting at most 20 ms for
+    // new events between each batch.
+    //
+    // The `&mut shared_data` is a mutable reference that will be forwarded to all
+    // your callbacks, allowing them to easily share some state
+    event_loop.run(Duration::from_millis(20), &mut shared_data, |shared_data| {
+        /*
+        * Insert here the processing you need to do do between each waiting batch (if any)
+        * like invoking your drawing logic if you're doing a GUI app for example.
+        */
+    });
+}
+```
+
+Calloop provides a few implementation of the `EventSource` trait such as a timer, a channel, and a
+generic adapter for monitoring a file descriptor for read/write readiness. At the moment calloop
+only supports Linux and the *BSD systems, but support for other platforms would be accepted if
+contributed!
+
+## Can it work alongside async/await?
+
+Since this new version, yes! Even is the core of your app works best will callbacks, some part of your
+needs might benefit from async/await. Since this last version, calloop allows you to mix both paradigms.
```

Comment:
'Since this last version' should probably be 'With this new version' again, or 'Since 0.7', or 'With the latest release'. Either way not 'last' (otherwise it kinda sounds like there will be no new versions after this one) and 'Since this (new|latest) version' sounds weird too (can't explain why though, sorry :sweat_smile:).